version.py

```diff
@@ -117,3 +123,15 @@ def find_latest_version_with_matching_major(reference_version: semver.VersionInf
             if not latest_candidate or latest_candidate < candidate:
                 latest_candidate = candidate
     return latest_candidate
+
+
+def partition_by_major_and_minor(
+    versions: Iterable[semver.VersionInfo],
+) -> Iterable[Set[semver.VersionInfo]]:
+    '''partition an iterable of semver VersionInfos by their joined major and minor version
```

Comment:
partition an iterable [..] into sorted sequences of VersionInfo, grouped by equal major and minor components. Sorting is done in ascending order, according to semver semantics. Duplicates are discarded.